src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -543,15 +544,26 @@ void InterpreterDropImpl::validate(const InterpreterDropImpl::TQuery & /*query*/
 ASTs InterpreterDropImpl::getRewrittenQueries(
     const InterpreterDropImpl::TQuery & drop_query, ContextPtr context, const String & mapped_to_database, const String & mysql_database)
 {
-    const auto & database_name = resolveDatabase(drop_query.getDatabase(), mysql_database, mapped_to_database, context);
-
-    /// Skip drop database|view|dictionary
-    if (database_name != mapped_to_database || !drop_query.table || drop_query.is_view || drop_query.is_dictionary)
+    /// Skip drop database|view|dictionary|others
+    if (drop_query.kind != TQuery::Kind::Table)
         return {};
-
-    ASTPtr rewritten_query = drop_query.clone();
-    rewritten_query->as<ASTDropQuery>()->setDatabase(mapped_to_database);
-    return ASTs{rewritten_query};
+    TQuery::QualifiedNames tables = drop_query.names;
+    ASTs rewritten_querys;
+    for (const auto & table: tables)
+    {
+        const auto & database_name = resolveDatabase(table.schema, mysql_database, mapped_to_database, context);
+        if (database_name != mapped_to_database)
+            continue;
+        auto rewritten_query = std::make_shared<ASTDropQuery>();
+        rewritten_query->setTable(table.shortName);
+        rewritten_query->setDatabase(mapped_to_database);
+        if (!drop_query.is_drop)
+            rewritten_query->kind = ASTDropQuery::Kind::Truncate;
+        rewritten_query->is_view = false;
+        rewritten_query->if_exists = drop_query.if_exists;
```

Comment:
Does it make sense to always set `rewritten_query->if_exists` to true? The problem is that it's impossible to drop multiple tables atomically in ClickHouse. If ClickHouse failed in the middle of operation for some reason, then if probable will retry. But retry will fail because some table does not exist, because it was dropped on previous attempt

Reply:
ok. This is a good behavior